### DISCUSSION TITLE
Tornare da ESPHome al firmware originale Sonoff

### DISCUSSION THREAD

**Discussion by Eddy16391** [indomus/forum#1938] [Hardware & Firmware]
Ciao a tutti,
come da titolo, ho due Sonoff Mini R2 su cui ho installato il firmware ESPHome.
Ora volevo sapere, è possibile tornare indietro? Volendo ho anche degli altri Sonoff Mini R2 con firmware originale da cui poter prendere un eventuale backup.
Fatemi sapere, grazie :-D

**Comment by indomus** [comment]
> 

se ti sei fatto un backup, certamente sì

**Reply by Eddy16391** [reply]
Il backup non l'ho fatto. Però potrei farlo partendo da un altro Sonoff in cui c'è ancora il firmware originale?
E, se si, dov'è la guida per fare queste cose? Non sono riuscito a trovarla.
Grazie :-D

**Comment by indomus** [comment]
https://indomus.it/guide/backup-restore-e-cancellazione-del-firmware-di-un-sonoff-e-simili/
https://indomus.it/guide/programmare-il-firmware-di-un-sonoff-e-simili-via-tasmotizer/

**Reply by Eddy16391** [reply]
Ok, queste le ho provate ieri. Ma è possibile che sui Sonoff con ESPHome installato non funzioni? Perchè ho saldato i pin come dice la guida e usato l'adattatore che viene letto correttamente dal pc (gli viene assegnata la COM3). Però qualsiasi comando provo a lanciare con esptool non riesce a fare la connessione.

**Reply by snakuzzo** [reply]
Si vede che non mandi correttamente in flash mode il sonoff.
Ricorda poi, che se usi lo stesso fw preso da un altro sonoff, se utilizzerai app ewelink, ne potrai vedere solo uno. Il firmware è unico (o almeno fino a qualche anno fa era così)

**Reply by Eddy16391** [reply]
Ah ok, e allora continuo ad usarli con esphome.
Grazie delle info
